### PR TITLE
Just models and specs for form tag validation.

### PR DESCRIPTION
**Pull request by mcantor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,29 +1,40 @@
...
-- Actions
-- Form Views
+## Basic Routes & Controllers
 
-make sure they are redirectnig and rendering correctly
-make sure they are calling save and update correctly
-make sure they print out full messages for an error summary on invalid forms
-fields should be prefilled
-error class on inputs
+1. Create `show`, `new`, `edit`, `create`, and `update` routes for both models.
```
Comment (ahimmelstoss) [inline README.md]: I would mention that they're creating controllers as well. the first error when running the specs is uninitialized AuthorsController.

File: README.md [hunk 2]
```diff
@@ -1,29 +1,40 @@
...
+- Apply an error class to a field conditionally based on errors on a field
 
-## Notes
+# Requirements
 
-Just give them the models and migrations.
+## Validations
 
-They have to add
+1. Add validations to `Author` such that...
```
Comment (ahimmelstoss) [inline README.md]: I think we should have simple model validation specs for these.

File: Gemfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+source 'https://rubygems.org'
```
Comment (ahimmelstoss) [inline Gemfile]: i think we should include `pry` as well.

File: spec/controllers/authors_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+require "rails_helper"
+
+RSpec.describe AuthorsController do
+  let(:author) { Author.create!(name: "H. Jeff", email: "jeff@sbahj.info", phone_number: "5553059028") }
+
+  describe "showing an author" do
+    it "shows an author" do
+      get :show, id: author.id
+      expect(assigns(:author)).to eq(author)
```
Comment (ahimmelstoss) [inline spec/controllers/authors_controller_spec.rb]: regarding what @loganhasson said about `assigns` https://github.com/learn-co-curriculum/validations-in-controller-actions-rails-lab/pull/1#discussion_r47662102 i think the same applies here.

File: README.md [hunk 5]
```diff
@@ -1,29 +1,41 @@
...
+1. Add validations to `Post` such that...
+  1. `title` is not blank
+  1. `content` is at least 100 characters long
+  1. `category` is either `"Fiction"` or `"Non-Fiction"`
+
+## Basic Routes & Controllers
+
+1. Create controllers for both models.
+1. Create `show`, `new`, `edit`, `create`, and `update` routes for both models.
+1. Define controller actions for `show`, `new`, and `edit`.
```
Comment (jmburges) [inline README.md]: Hey @mcantor, so we already had students build the stuff like the the simple routes, actions and such in the last lab they did. So I would include the solution to the last lab they did in here as the starter code. Then the point of this lab is to just have them render those errors to students.

File: spec/controllers/authors_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    it "does not save" do
+      expect(@bad).to be_new_record
+    end
+
+    it "does not create" do
+      expect(Author.find_by(phone_number: bad_attributes[:phone_number])).to be_nil
+    end
+
+    it "has an error for missing name" do
+      expect(@bad.errors[:name]).to_not be_empty
```
Comment (jmburges) [inline spec/controllers/authors_controller_spec.rb]: Also I'd include some tests (Capybara may be easiest) that actually test that the student is rendering the errors on the form.

File: spec/controllers/posts_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  describe "making invalid updates" do
+    let(:bad_attributes) do
+      attributes.merge(
+        title: nil,
+        content: "too short",
+        category: "Speculative Fiction"
+      )
+    end
+
+    before { @article.update(bad_attributes) }
```
Comment (jmburges) [inline spec/controllers/posts_controller_spec.rb]: Also I'd include some tests (Capybara may be easiest) that actually test that the student is rendering the errors on the form.

File: spec/views/authors/new.html.erb_spec.rb [hunk 8]
```diff
@@ -0,0 +1,52 @@
+RSpec.describe "authors/new" do
+  before do
+    assign(:author, author)
```
Comment (jmburges) [inline spec/views/authors/new.html.erb_spec.rb]: no assign

File: spec/views/authors/new.html.erb_spec.rb [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  end
+
+  describe "a blank form" do
+    let(:author) { Author.new }
+
+    before(:each) { render }
+
+    it "does not render an error list" do
+      assert_select "#error_explanation", false
```
Comment (jmburges) [inline spec/views/authors/new.html.erb_spec.rb]: they've never seen `assert_select`. We generally do view layer tests with Capybara

**Comment by jmburges** [comment]
So for view layer tests we generally just do Capybara tests similar to this: https://github.com/learn-co-curriculum/rails-pollywog-ranch/blob/master/spec/03_features/metamorphosize_spec.rb. Also something like this: http://ruby-journal.com/how-to-write-rails-view-test-with-rspec/ would be good

**Comment by jmburges** [comment]
back at you!

**Pull request merged**